hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -1418,6 +1419,10 @@ protected void commitInternal(String instantTime, Map<String, HoodieData<HoodieR
     metrics.ifPresent(m -> m.updateSizeMetrics(metadataMetaClient, metadata, dataMetaClient.getTableConfig().getMetadataPartitions()));
   }
 
+  protected abstract void writeAndCommitBulkInsert(BaseHoodieWriteClient<?, I, ?, O> writeClient, String instantTime, I preppedRecordInputs, Option<BulkInsertPartitioner> bulkInsertPartitioner);
```

Comment:
Let's add inline compaction methods in write client and remove these methods.

Reply:
sure